packages/legend-graph/src/helpers/DomainHelper.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Class } from '../models/metamodels/pure/packageableElements/domain/Class';
+
+export const isVersionedClass = (val: Class): boolean => {
+  for (let i = 0; i < val.stereotypes.length; i++) {
+    const stereotype = val.stereotypes[i]?.value.value;
```

Comment:
Let's change the logic to be if `any` of the stereotypes for the given class has the profile `meta::pure::profiles` as the owner then its versioned. 


@YannanGao-gs Please look at this MR as some of the logic here will overlap with your work.